src/backend/columnar/sql/columnar--10.2-2--10.2-3.sql

```diff
@@ -0,0 +1,15 @@
+-- columnar--10.2-2--10.2-3.sql
+
+-- Since stripe_first_row_number_idx is required to scan a columnar table, we
+-- need to make sure that it is created before doing anything with columnar
+-- tables during pg upgrades.
+--
+-- However, a plain btree index is not a dependency of a table, so pg_upgrade
+-- cannot guarantee that stripe_first_row_number_idx gets created when
+-- creating columnar.stripe, unless we make it a unique "constraint".
+--
+-- To do that, drop stripe_first_row_number_idx and create a unique
+-- constraint with the same name to keep the code change at minimum.
+DROP INDEX columnar.stripe_first_row_number_idx;
+ALTER TABLE columnar.stripe ADD CONSTRAINT stripe_first_row_number_idx
```

Comment:
Is there any downside of making this index UNIQUE? i.e. could this cause any conflicting rows. Otherwise another approach would be to manually add an entry to pg_depend for the normal index.

Reply:
> i.e. could this cause any conflicting rows.

No it shouldn't, unless a bug happens there :) So it would be useful to detect such unexpected cases too.